test/fixtures/rules/std_rule_cases/L003.yml

```diff
@@ -1516,3 +1516,42 @@ test_fail_case_else_end_clause:
         end as field,
         bar
     from foo
+
+test_fail_fix_consistency_around_comments:
+  # Check that comments don't make fixes inconsistent.
+  # https://github.com/sqlfluff/sqlfluff/issues/4223
+  fail_str: |
+    select
+        case
+            when a
+            then b
+        end as foo,
+        case
+            when a -- bar
+            then b
+        end as bar
+    from c
+  fix_str: |
+    select
+        case
+            when a
+                then b
+        end as foo,
+        case
+            when a -- bar
+                then b
+        end as bar
+    from c
+
+test_fail_coverage_indent_trough:
+  # This test primarily tests the handling of closing trough indents
+  fail_str: |
+    WITH bar as (SELECT 1
```

Comment:
In the past, L003 only handled indentation. Any concerns that here it is moving code between lines?

Reply:
I don't think so - this is functionality that is already out in the wild in the treatment of existing hanging indents in `2.0.0a1`. All this PR does is fix a bug with it.